compile.sh

```diff
@@ -5,3 +5,4 @@ RUSTFLAGS="-Zlocation-detail=none" cargo +nightly build --manifest-path rust/Car
 kotlinc -include-runtime kotlin/code.kt -d kotlin/code.jar
 #kotlinc-native -include-runtime kotlin/code.kt -d kotlin/code
 dart compile exe dart/code.dart -o dart/code --target-os=macos
+ghc -O2 haskell/code.hs -o haskell/code
```

Comment:
Performance might be improved with additional `-fllvm` flag, which uses LLVM backend for compilation. It really matters in some cases according to the [documentation]. Here is my personal experiment: [link][benchmark]

Note: currently supported LLVM version is 13, 14, 15; compilation would fail otherwise.

```sh
if ghc -O2 -fllvm haskell/code.hs -o haskell/code; then
    true
else
    echo "ghc: cannot compile with llvm backend; fallback to use default backend"
    ghc -O2 haskell/code.hs -o haskell/code
fi
```

Or one liner:

```sh
ghc -O2 -fllvm haskell/code.hs -o haskell/code || { echo "ghc: cannot compile with llvm backend; fallback to use default backend"; ghc -O2 haskell/code.hs -o haskell/code; }
```

[documentation]: https://downloads.haskell.org/ghc/latest/docs/users_guide/codegens.html
[benchmark]: https://github.com/kimminss0/haskell-playground/tree/92fc63f15a41fcdd5c9260a9b4ed45554d2cc9a2/fibonacci/benchmark

Reply:
Tested on fibonacci:

with llvm 14:
```
% \time  ./code 40
165580140
        0.72 real         0.70 user         0.00 sys
```

without llvm; using default backend:
```
% \time  ./code 40
165580140
        0.93 real         0.91 user         0.00 sys
```

Reply:
Thanks for your suggestion! I have adopted it in latest commit.